Guide to Open Github Issue (#2204)
port-labs/port-docs: docs/guides/all/open-github-issue.md

---
displayed_sidebar: null
description: Learn how to create a GitHub issue in Port, streamlining issue categorization and tracking.
---

import ExistingSecretsCallout from '/docs/guides/templates/secrets/_existing_secrets_callout.mdx'
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

# Create a Github issue

## Overview
This guide will help you implement a self-service action in Port that allows you to create GitHub issues directly from Port using **synced webhooks**.
This functionality streamlines project management by enabling users to create issues without leaving Port.

## Prerequisites

- Complete the [onboarding process](/getting-started/overview).
- Access to your GitHub organization with permissions to manage issues.
- Optional - Install Port's [GitHub integration](https://docs.port.io/build-your-software-catalog/sync-data-to-catalog/git/github/).

## Set up data model

If you haven't installed the GitHub integration, you will need to manually create a blueprint for GitHub repository.  
We highly recommend that you install the GitHub integration to have such resources automatically set up for you. 

> ### Create a GitHub repository blueprint

Reviewer: ```suggestion
<h3> Create the repository blueprint </h3>
```

1. Go to your [Builder](https://app.getport.io/settings/data-model) page.
2. Click on `+ Blueprint`.
3. Click on the `{...}` button in the top right corner, and choose `Edit JSON`.

> 4. Add the following JSON schema into the editor while clicking `Save` to create the blueprint:

Reviewer: Can we separate the `Save` step into a 5th step to make the Instruction more clear and specific

            "title": "README",
            "type": "string",
            "format": "markdown"
        },
        "url": {
            "title": "Repository URL",
            "type": "string",
            "format": "url"
        },
        "defaultBranch": {
            "title": "Default branch",
            "type": "string"
        }
        },
        "required": []
    },
    "mirrorProperties": {},
    "calculationProperties": {},
    "relations": {}
    }
    ```
    </details>

## Implementation

You can create GitHub issue by leveraging Port's **synced webhooks** and **secrets** to directly interact with GitHub's REST API.

### Add Port secrets

To add these secrets to your portal:

1. Click on the `...` button in the top right corner of your Port application.

2. Click on **Credentials**.

3. Click on the `Secrets` tab.

4. Click on `+ Secret` and add the following secrets:
    - `GITHUB_TOKEN`: Your [GitHub Personal Access Token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens) with the permission to create issue.
    - `GITHUB_ORG`: Your GitHub organization or username

### Set up self-service action

> Let's define a self-service action that is used to create a GitHub issue:

Reviewer: ```suggestion
Follow these steps to create the self-service action:
```
